service/history/shard/context_impl.go

```diff
@@ -1327,7 +1342,7 @@ func (s *ContextImpl) allocateTaskIDsLocked(
 			// if scheduled task, check if fire time is in the past
 			if category.Type() == tasks.CategoryTypeScheduled {
 				ts := task.GetVisibilityTime()
-				if task.GetVersion() != common.EmptyVersion {
+				if task.GetVersion() != common.EmptyVersion && category.ID() == tasks.CategoryIDTimer {
```

Comment:
This prevent other single processor schedule queue from running into the same problem. and for those scheduled queues, they always need to pass `false` as the flag when getting the max read level (kinda hacky). 

This check plus the logic added in this PR can all be removed in 1.19 when removing the old task processing logic.